### DISCUSSION TITLE
Mantine v7 support

### DISCUSSION THREAD

**Discussion by skelawsky** [KevinVandy/mantine-react-table#68] [Q&A]
Mantine v7 will releases soon. Will mantine-react-table work the same without deprecated createStyles func.?

**Comment by KevinVandy** [comment]
Last I heard, the creator of Mantine said that it would come out in October. But yes there will be a larger refactor by then.

**Reply by DanHorrocksBurgess** [reply]
Looks like Mantine v7 entered Beta (All be it beta 0!) yesterday:

https://github.com/mantinedev/mantine-v7/releases/tag/7.0.0-beta.0

**Comment by tekfunk** [comment]
Well I got it 'working' with 7, but with no styles (obviously) and it won't build due to some dependency change. something got moved out of mantine/core I think

**Reply by KevinVandy** [reply]
It's not going to really be compatible with v7 until a major refactor. I might be starting it in the next couple of weeks though.

**Reply by DenisBessa** [reply]
Mantine 7.0.0 just got out! @KevinVandy let us know if we can help in the migration process. I've already done it in my app, so maybe I can help with something.

**Reply by paulm17** [reply]
@DenisBessa Was about to post this as well.  Keep us posted in the mantine discord @KevinVandy

**Reply by KevinVandy** [reply]
I'd love help with the upgrade. Making a bunch of the base changes needed to the repo today for the work needed

**Comment by DenisBessa** [comment]
Looking forward to use with Mantine v7.

**Comment by KevinVandy** [comment]
@DenisBessa @paulm17 git branches are now v1 and v2.

Also made a https://v2.mantine-react-table.com/ subdomain. No upgrade work has actually been done yet though.

**Comment by KevinVandy** [comment]
https://github.com/KevinVandy/mantine-react-table/discussions/169

**Comment by rommelmamedov** [comment]
@KevinVandy do we have a list of tasks/todos that need to be done for v7 migration? I can help with it.

**Reply by KevinVandy** [reply]
@rommelmamedov  sorry I never saw this message. Join the discord if you are interested in contributing. There are a handful of community contributors now and that's where we coordinate.
We're getting close to going from alpha to beta. Most of the remaining work is in CSS standardization, and then updating hundreds of examples (not exaggerating lol) and rewriting more of the docs.

**Comment by lukasbash** [comment]
Cool work for the table! Is there by chance some timeline or roadmap available when you plan to have a first stable release on v7?

**Reply by mamlzy** [reply]
@KevinVandy

**Reply by KevinVandy** [reply]
> @KevinVandy 

See the v2 docs and discord. Lots of development still happening

**Reply by mamlzy** [reply]
thanks for your response!

**Comment by danikane** [comment]
This just in:
https://mantine.dev/guides/6x-to-7x/#migration-to-mantineemotion
https://github.com/mantinedev/mantine/releases/tag/7.9.0

Am I right to assume this simplifies things for MRT as well?

**Reply by alessandrojcm** [reply]
MRT doesn't use emotion in V2 so this does not affect it.

**Reply by KevinVandy** [reply]
I haven't looked at the v7 emotion use yet, but it may make using the sc prop possible again with some configuration
